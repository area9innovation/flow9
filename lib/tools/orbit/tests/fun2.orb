fn quote(a : ast) = a;
let g = makeOGraph("graph");
let f = quote(\x -> 2*x);
let n = addOGraphWithSub(g, f, []);

matchOGraphPattern(g, quote(`\`), \(b,e) -> (
	println(extractOGraph(g,e));
));

println(extractOGraph(g, n));
f is (\a.b => Lambda(a, b)); // Matches a lambda with one arg
