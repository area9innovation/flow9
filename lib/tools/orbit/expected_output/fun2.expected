Î»x.(2 * x)
Pattern matched
(2 * x)
Result: 1
